src/slam/src/slam_solver/graph_slam_solver/graph_slam_solver.cpp

```diff
@@ -129,6 +131,20 @@ void GraphSLAMSolver::add_observations(const std::vector<common_lib::structures:
   association_time = rclcpp::Clock().now();
   RCLCPP_DEBUG(rclcpp::get_logger("slam"), "add_observations - Associations calculated");
 
+  Eigen::Vector3d pose;
+  pose << state(0), state(1), state(2);
+
+  Eigen::VectorXi landmarks(state.size() - 3);
+  for (int i = 3; i < state.size(); i ++) {
+    // Iterate through the landmarks in the state
+      landmarks(i-3) = state(i);
+  }
```

Comment:
This can be done directly with segment function or something like that, check the rest of the file for examples